app/controllers/vm_cloud_controller.rb

```diff
@@ -179,6 +179,81 @@ def live_migrate_vm
     end
   end
 
+  def evacuate
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id]) # Set the VM object
+    drop_breadcrumb(
+      :name => _("Evacuate Instance '%{name}'") % {:name => @record.name},
+      :url  => "/vm_cloud/evacuate"
+    ) unless @explorer
+    @in_a_form = true
+    @refresh_partial = "vm_common/evacuate"
+  end
+  alias_method :instance_evacuate, :evacuate
+
+  def evacuate_form_fields
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id])
+    clusters = []
+    hosts = []
+    @record.ext_management_system.find_filtered_children("ems_clusters").each do |c|
+      clusters << {:id => c.id, :name => c.name}
+    end
+    @record.ext_management_system.find_filtered_children("hosts").each do |h|
+      hosts << {:id => h.id, :name => h.name, :cluster_id => h.emd_cluster.id}
+    end
+    clusters.sort
+    hosts.sort
+    render :json => {
+      :clusters => clusters,
+      :hosts    => hosts
+    }
+  end
+
+  def evacuate_vm
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id])
+
+    case params[:button]
+    when "cancel"
+      add_flash(_("Evacuation of %{model} \"%{name}\" was cancelled by the user") % {
+        :model => ui_lookup(:table => "vm_cloud"), :name => @record.name})
+      session[:flash_msgs] = @flash_array.dup
+      render :update do |page|
+        page.redirect_to(previous_breadcrumb_url)
```

Comment:
@mansam You need to add -
```
page << javascript_prologue
```
when you are rendering a page, otherwise the `render` will not work.  (This change was recently added)
If you grep `render :update do |page|`, you will see that all instances of page render now have the `javascript_prologue` bit added.